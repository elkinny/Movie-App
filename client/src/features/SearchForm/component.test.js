import SearchFormComponent from './component';
import React from 'react';
import { shallow } from 'enzyme';

describe('SearchFormComponent: ', () => {
  const handleSubmit = jest.fn();
  const props = {
    handleSubmit: handleSubmit,
    handleInput: () => {},
    searchToggleValue: 'searchToggleValue',
    searchValue: 'searchValue',
  };

  let component;

  beforeEach(() => {
    component = shallow(<SearchFormComponent {...props} />);
  });

  it('1. To match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('2. To call handleSubmit on submit form', () => {
    component.find('form').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
