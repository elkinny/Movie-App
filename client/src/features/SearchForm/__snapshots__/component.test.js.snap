// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFormComponent:  1. To match snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
  <Toggle
    defaultValue={true}
    labels={
      Array [
        "Title",
        "Genre",
      ]
    }
    onChange={[Function]}
    text="Search by:"
    type="search"
    values={
      Array [
        "title",
        "genres",
      ]
    }
  />
  <input
    type="submit"
    value="Search"
  />
</form>
`;
